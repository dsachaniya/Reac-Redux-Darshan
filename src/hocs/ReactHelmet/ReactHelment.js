import React from 'react';
import { Helmet } from 'react-helmet-async';

/**
 * ReactHelmet: Dynamically loads page title and description
 * @param {Node} WrappedComponent
 * @param {string} title
 * @param {string} description
 * @returns {Function}
 */
function ReactHelmet(WrappedComponent, title, description) {
  return function ReactHelmetComponent(props) {
    return (
      <>
        <Helmet>
          <title>{title}</title>
          <meta name="description" content={description} />
        </Helmet>
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <WrappedComponent {...props} />
      </>
    );
  };
}

export default ReactHelmet;
