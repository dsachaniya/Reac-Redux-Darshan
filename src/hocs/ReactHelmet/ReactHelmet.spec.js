import React from 'react';
import { shallow } from 'enzyme';
import ReactHelment from './ReactHelment';

/**
 * TestComponent
 * @returns {Node}
 */
function TestComponent() {
  return <div>TEST</div>;
}

describe('<ReactHelment />', () => {
  it('Rendering', () => {
    // 1. Call hoc function
    const Component = ReactHelment(
      TestComponent,
      'Test - React - Redux Darshans',
      'Test to be protected.',
    );
    const wrapper = shallow(<Component />);

    // 2. Snapshot
    expect(wrapper).toMatchSnapshot();

    // 3. Test it contains 'title' and 'meta' tags
    expect(wrapper.find('title')).toHaveLength(1);
    expect(wrapper.find('meta')).toHaveLength(1);

    // 4. Test it renders with proper title and description
    expect(wrapper.find('title').text()).toEqual(
      'Test - React - Redux Darshans',
    );
    expect(wrapper.find('meta').prop('content')).toEqual(
      'Test to be protected.',
    );
  });
});
