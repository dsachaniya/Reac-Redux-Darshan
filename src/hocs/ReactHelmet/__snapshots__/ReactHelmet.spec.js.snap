// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactHelment /> Rendering 1`] = `
<Fragment>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Test - React - Redux Darshans
    </title>
    <meta
      content="Test to be protected."
      name="description"
    />
  </Helmet>
  <TestComponent />
</Fragment>
`;
